src/Language/Rust.ts

```diff
@@ -0,0 +1,312 @@
+            _anyType => maybeAnnotated(withIssues, anyTypeIssueAnnotation, "serde_json::Value"),
+            _nullType => maybeAnnotated(withIssues, nullTypeIssueAnnotation, "Option<serde_json::Value>"),
+            _boolType => "bool",
+            _integerType => "i64",
+            _doubleType => "f64",
+            _stringType => "String",
+            arrayType => ["Vec<", this.rustType(arrayType.items, withIssues), ">"],
+            classType => this.nameForNamedType(classType),
+            mapType => ["HashMap<String, ", this.rustType(mapType.values, withIssues), ">"],
+            enumType => this.nameForNamedType(enumType),
+            unionType => {
+                const nullable = nullableFromUnion(unionType);
+
+                if (nullable) return this.nullableRustType(nullable, withIssues);
+
+                const [hasNull,] = removeNullFromUnion(unionType);
+
+                const isCycleBreaker = this.isCycleBreakerType(unionType);
+
+                const name = isCycleBreaker ? ["Box<", this.nameForNamedType(unionType), ">"] : this.nameForNamedType(unionType);
+
+                return hasNull
+                    ? ["Option<", name, ">"] as Sourcelike
+                    : name;
+            }
+        );
+    };
+
+    private breakCycle = (t: Type, withIssues: boolean): any => {
+        const rustType = this.rustType(t, withIssues);
+        const isCycleBreaker = this.isCycleBreakerType(t);
+
+        return isCycleBreaker ? ["Box<", rustType, ">"] : rustType;
+    }
+
+    private emitStructDefinition = (c: ClassType, className: Name): void => {
+        this.emitLine("#[derive(Serialize, Deserialize)]");
+
+        const structBody = () => this.forEachClassProperty(c, "none", (name, jsonName, prop) => {
+
+            const escapedName = rustStringEscape(jsonName);
+
+            this.emitLine("#[serde(rename = \"", escapedName, "\")]");
+
+            this.emitLine(name, ": ", this.breakCycle(prop.type, true), ",");
+        });
+
+        this.emitBlock(["pub struct ", className], structBody);
+    };
+
+    private emitBlock = (line: Sourcelike, f: () => void): void => {
+        this.emitLine(line, " {");
+        this.indent(f);
+        this.emitLine("}");
+    };
+
+    private emitUnion = (u: UnionType, unionName: Name): void => {
+        const isMaybeWithSingleType = nullableFromUnion(u);
+
+        if (isMaybeWithSingleType) {
+            return;
+        }
+
+        this.emitLine("#[derive(Serialize, Deserialize)]");
+        this.emitLine("#[serde(untagged)]");
+
+        const [, nonNulls] = removeNullFromUnion(u);
+
+        const unionBody = () => this.forEachUnionMember(u, nonNulls, "none", null, (fieldName, t) => {
+            const rustType = this.breakCycle(t, true);
+            this.emitLine([fieldName, "(", rustType, "),"]);
+        });
+
+        this.emitBlock(["pub enum ", unionName], unionBody);
+    }
+
+    emitEnumDefinition = (e: EnumType, enumName: Name): void => {

```

Comment:
Don't bother fixing it, but for future reference: We don't declare methods like that anymore, i.e. with `=`.  We haven't changed all of them in the codebase yet, sorry.

Short explanation: `=` is slower and uses more memory because it allocates a closure, so in speed-sensitive circumstances we have to use the non-`=` way, and I don't want to have two different ways of defining methods, because that causes confusion is a pain to maintain.

Reply:
Thank you for the explanation.